earth2grid/lcc.py

```diff
@@ -0,0 +1,201 @@
+)
+
+
+class LambertConformalConicGrid(base.Grid):
+    # nothing here is specific to the projection, so could be shared by any projected rectilinear grid
+    def __init__(self, projection: LambertConformalConicProjection, x, y):
+        """        
+        Args:
+            projection: LambertConformalConicProjection object
+            x: range of x values
+            y: range of y values
+
+        """
+        self.projection = projection
+
+        self.x = np.array(x)
+        self.y = np.array(y)
+
+    @property
+    def lat_lon(self):
+        mesh_x, mesh_y = np.meshgrid(self.x, self.y)
+        return self.projection.inv(mesh_x, mesh_y)
+
+    @property
+    def lat(self):
+        return self.lat_lon[0]
+
+    @property
+    def lon(self):
+        return self.lat_lon[1]
+
+    @property
+    def shape(self):
+        return (len(self.y), len(self.x))
+
+    def __getitem__(self, idxs):
+        yidxs, xidxs = idxs
+        return LambertConformalConicGrid(self.projection, x=self.x[xidxs], y=self.y[yidxs])
+
+    def get_bilinear_regridder_to(self, lat: np.ndarray, lon: np.ndarray):
+        """Get regridder to the specified lat and lon points"""
+        return _RegridFromLCC(self, lat, lon)
+
+    def visualize(self, data):
+        raise NotImplementedError()
+
+    def to_pyvista(self):
+        if pv is None:
+            raise ImportError("Need to install pyvista")
+
+        lat, lon = self.lat_lon
+        y = np.cos(np.deg2rad(lat)) * np.sin(np.deg2rad(lon))
+        x = np.cos(np.deg2rad(lat)) * np.cos(np.deg2rad(lon))
+        z = np.sin(np.deg2rad(lat))
+        grid = pv.StructuredGrid(x, y, z)
+        return grid
+
+def hrrr_conus_grid(ix0 = 0, iy0 = 0, nx = 1799, ny = 1059):
+    # coordinates of point in top-left corner
+    lat0 = 21.138123
+    lon0 = 237.280472
+    # grid length (m)
+    scale = 3000.0
+    # coordinates on projected space
+    x0, y0 = HRRR_CONUS_PROJECTION.proj(lat0, lon0)
+
+    x = [x0 + i * scale for i in range(ix0, ix0+nx)]
+    y = [y0 + i * scale for i in range(iy0, iy0+ny)]
+
+    return LambertConformalConicGrid(HRRR_CONUS_PROJECTION, x, y)
+
+# Grid used by HRRR CONUS (Continental US) data
+HRRR_CONUS_GRID = hrrr_conus_grid()
+
+
+class _RegridFromLCC(torch.nn.Module):
+    """Regrid from LambertConformalConicGrid to unstructured grid with bilinear interpolation"""
+
+    def __init__(self, src: LambertConformalConicGrid, lat: np.ndarray, lon: np.ndarray):
+        super().__init__()
+
+        x, y = src.projection.proj(lat, lon)
+        
+        self.shape = lat.shape
+        self._bilinear = BilinearInterpolator(
+            x_coords = torch.from_numpy(src.x), 
+            y_coords = torch.from_numpy(src.y), 
+            x_query = torch.from_numpy(x.ravel()),
+            y_query = torch.from_numpy(y.ravel()))
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        out = self._bilinear(x)
+        return out.view(out.shape[:-1] + self.shape)
```

Comment:
Is this wrapper class (_RegridFromLCC) necessary? I thought this reshaping behavior was already supported by BilinearInterpolator.

Reply:
BilinearInterpolator wasn't working for multidimensional destination grids, but I think I've fixed that now.